### PR TITLE
Add a persistent side-by-side ratio setting

### PR DESCRIPTION
**Pull request by echarles**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/12157

## Code changes

Add a `sideBySideOutputRatio` in the notebook setting and use that value in the popup form when user wants to change the ratio via the command palette.

## User-facing changes

Untitled5

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by echarles** [comment]
CI is green. This is ready for review.

File: packages/notebook-extension/schema/tracker.json [hunk 1]
```diff
@@ -922,6 +922,12 @@
       "description": "Side-by-side right margin override.",
       "type": "string",
       "default": "10px"
+    },
+    "sideBySideOutputRatio": {
+      "title": "Side-by-side output ratio",
+      "description": "For the side-by-side rendering, the side-by-side output ratio defines the width of the output vs the input. Set 1 for same size, > 1 for larger output, < 1 for smaller output.",
+      "type": "number",
+      "default": 1
```
Comment (fcollonval) [inline packages/notebook-extension/schema/tracker.json:930]: Minor enhancement:

```suggestion
      "default": 1,
      "exclusiveMinimum": 0
```

**Comment by echarles** [comment]
@fcollonval I have merged your suggestion and now it was failing with

```
[W 2022-05-24 08:27:47.689 BrowserApp] Failed validating schema (@jupyterlab/notebook-extension:tracker): 0 is not of type 'boolean'
    
    Failed validating 'type' in metaschema['properties']['properties']['additionalProperties']['properties']['exclusiveMinimum']:
        {'default': False, 'type': 'boolean'}
    
    On schema['properties']['sideBySideOutputRatio']['exclusiveMinimum']:
        0
```


So I made it a boolean, and now I have another error:

````
[W 2022-05-24 08:46:20.558 ServerApp] 500 GET /foo/lab/api/settings?1653381980433 (127.0.0.1): Failed validating schema (@jupyterlab/notebook-extension:tracker): 'minimum' is a dependency of 'exclusiveMinimum'

...
   On schema['properties']['sideBySideOutputRatio']:
        {'default': 1,
         'description': 'For the side-by-side rendering, the side-by-side '
                        'output ratio defines the width of the output vs the '
                        'input. Set 1 for same size, > 1 for larger output, < '
                        '1 for smaller output.',
         'exclusiveMinimum': True,
         'title': 'Side-by-side output ratio',
```

**Comment by fcollonval** [comment]
That means we are using Draft 4 :sob: 
So I advice to bump ajv to the latest version and stick with `exclusiveMinimum: number`. But if you plan to backport this one, I would argue not to bump ajv as it will be on a patch release. And therefore you need to set `minimum: number` and `exclusiveMinimum: boolean` - see: https://json-schema.org/understanding-json-schema/reference/numeric.html

**Comment by echarles** [comment]
> That means we are using Draft 4 😭

The reported errors are from the GitHub CI jobs.

**Comment by echarles** [comment]
> So in fact you need to set the exclusiveMinimum to a boolean (and therefore set minimum) because the JSON schema constrained comes from jupyterlab_server:

I have pushed and the previous errors dissapear. CI was green, but with that changes it is constantly failing on 5+ flows and I can make sense of what is going wrong based on the log. 

```
      "minimum": 0,
      "exclusiveMinimum": true
```

**Comment by fcollonval** [comment]
I suspect the json schema validator is not consistent at some place and so it uses either draft04 at some place and more recent version at others. So I would suggest setting only `minimum` attribute.

**Comment by echarles** [comment]
@fcollonval I have pushed your suggestion. I see now 2 failing tests, which are unrelated IMHO.

**Comment by fcollonval** [comment]
Thanks @echarles 
Do you know why this PR is showing so much changes? Would you mind to rebase it?

**Comment by echarles** [comment]
This branch has issues. Closing in favor of follow-up https://github.com/jupyterlab/jupyterlab/pull/12633

**Pull request closed without merging**